gridData/OpenDX.py

```diff
@@ -182,18 +192,118 @@ def write(self,file):
                       ('counts '+self.ndformat(' %d')) % tuple(self.shape))
 
 class array(DXclass):
-    """OpenDX array class"""
-    def __init__(self,classid,array=None,**kwargs):
+    """OpenDX array class.
+
+    See `Array Objects`_ for details.
+
+    .. _Array Objects:
+       https://web.archive.org/web/20080808140524/http://opendx.sdsc.edu/docs/html/pages/usrgu068.htm#Header_440
+    """
+    # equivalence between DX types and numpy dtypes.name
+    # (round-tripping is not guaranteed to produce identical types)
+    np_types = {
+        "uint8": "byte",         # DX "unsigned byte" equivalent
+        "int8": "signed byte",
+        "uint16": "unsigned short",
+        "int16": "short",         # DX "signed short" equivalent
+        "uint32": "unsigned int",
+        "int32": "int",           # DX "signed int"   equivalent
+        "uint64": "unsigned int", # not explicit in DX, for compatibility
+        "int64": "int",           # not explicit in DX, for compatibility
+        # "hyper",                # ?
+        "float32": "float",       # default
+        "float64": "double",
+        "float16": "float",       # float16 not available in DX, use float
+        # numpy "float128 not available, raise error
+        # "string" not automatically supported
+    }
+    dx_types = {
+        "byte": "uint8",
+        "unsigned byte": "uint8",
+        "signed byte": "int8",
+        "unsigned short": "uint16",
+        "short": "int16",
+        "signed short": "int16",
+        "unsigned int": "uint32",
+        "int": "int32",
+        "signed int": "int32",
+        "unsigned int": "uint64", # not explicit in DX, for compatibility
+        "int": "int64",           # not explicit in DX, for compatibility
+        # "hyper",                # ?
+        "float": "float32",       # default
+        "double": "float64",
+        # "string" not automatically supported
+    }
+
+    def __init__(self, classid, array=None, type=None, **kwargs):
+        """
+        Parameters
+        ----------
+        classid : int
+        array : array_like
+        type : str (optional)
```

Comment:
I would prefer to name this `dtype` for **D**ata**Type** following the numpy naming. That way we also avoid clashing with the buildin `type`.

Reply:
dtype is misleading because they are not numpy.dtypes. They are the specific labels in the DX file format specification. But "dxtype" seemed clunky and I didn't see a problem in shadowing the builtin type if it was clearer to the user. I also did not want to use dxtype as argument for export because if we ever want to use the data type for other formats then we can parse type appropriately.